manifests/init.pp

```diff
@@ -1,25 +1,62 @@
 # This class configures, installs, and ensures GDM is running.
 #
+# @param dconf_hash
+#   ``dconf`` settings applicable to GDM
+#
+#  @see dconf(5)
+#  @see data/common.yaml
+#
+# @param packages
+#   A Hash of packages to be installed
+#
+#   * NOTE: Setting this will *override* the default package list
+#   * The ensure value can be set in the hash of each package, like the example
+#     below:
+#
+#   @example Override packages
+#     { 'gdm' => { 'ensure' => '1.2.3' } }
+#
+#   @see data/common.yaml
+#
+# @param package_ensure
+#   The SIMP global catalyst to set the default `ensure` settings for packages
+#   managed with this module.
+#
 # @param include_sec Boolean
-# Whether or not to include gdm::sec
+#   Include gdm::sec security settings
 #
-# @author Trevor Vaughan <tvaughan@onyxpoint.com>
-# @author Nick Markowski <nmarkowski@keywcorp.com>
+# @param auditd
+#   Enable auditd support for this module via the ``simp-auditd`` module
+#
+# @author https://github.com/simp/pupmod-simp-gdm/graphs/contributors
 #
 class gdm (
-  Boolean $include_sec = true,
-  Boolean $auditd      = simplib::lookup('simp_options::auditd', { 'default_value' => false })
+  Gnome::DconfSettings            $dconf_hash,
+  Hash[String[1], Optional[Hash]] $packages,
+  Simplib::PackageEnsure          $package_ensure = simplib::lookup('simp_options::package_ensure', { 'default_value' => 'installed' }),
+  Boolean                         $include_sec    = true,
+  Boolean                         $auditd         = simplib::lookup('simp_options::auditd', { 'default_value'         => false }),
 ) {
 
-  include '::gdm::install'
+  simplib::assert_metadata($module_name)
+
+  gnome::install { 'gdm packages':
```

Comment:
Seems like this should be in simplib.  Don't like the unnecessary coupling with gnome module.

Reply:
@lnemsick-simp We can move it later. I wanted to talk about potentially doing that since it could be a bit weird. GDM needs gnome for other things anyway so it's not a huge issue.